extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/VaultSystemBackendManager.java

```diff
@@ -0,0 +1,106 @@
+import io.quarkus.vault.runtime.config.VaultBuildTimeConfig;
+import io.quarkus.vault.runtime.sys.health.VaultHealth;
+import io.quarkus.vault.runtime.sys.health.VaultHealthStatus;
+import io.quarkus.vault.runtime.sys.seal.VaultInit;
+import io.quarkus.vault.runtime.sys.seal.VaultSealStatus;
+
+public class VaultSystemBackendManager implements VaultSystemBackendEngine {
+
+    private VaultClient vaultClient;
+    private VaultBuildTimeConfig buildTimeConfig;
+
+    public VaultSystemBackendManager(VaultBuildTimeConfig buildTimeConfig, VaultClient vaultClient) {
+        this.vaultClient = vaultClient;
+        this.buildTimeConfig = buildTimeConfig;
+    }
+
+    @Override
+    public VaultInit init(int secretShares, int secretThreshold) {
+        final VaultInitResponse init = this.vaultClient.init(secretShares, secretThreshold);
+
+        final VaultInit vaultInit = new VaultInit(init.keys, init.keysBase64, init.rootToken);
+        return vaultInit;
+    }
+
+    @Override
+    public VaultHealth health() {
+
+        boolean isStandByOk = false;
+        if (this.buildTimeConfig.health.standbyok) {
+            isStandByOk = true;
+        }
+
+        boolean isPerfStandByOk = false;
+        if (this.buildTimeConfig.health.perfstandbyok) {
+            isPerfStandByOk = true;
+        }
+
+        return this.health(isStandByOk, isPerfStandByOk);
+    }
+
+    @Override
+    public VaultHealthStatus healthStatus() {
+        boolean isStandByOk = false;
+        if (this.buildTimeConfig.health.standbyok) {
+            isStandByOk = true;
+        }
+
+        boolean isPerfStandByOk = false;
+        if (this.buildTimeConfig.health.perfstandbyok) {
+            isPerfStandByOk = true;
+        }
+
+        return this.healthStatus(isStandByOk, isPerfStandByOk);
+    }
+
+    @Override
+    public VaultSealStatus sealStatus() {
+        final VaultSealStatusResult vaultSealStatusResult = this.vaultClient.systemSealStatus();
+
+        final VaultSealStatus vaultSealStatus = new VaultSealStatus();
+        vaultSealStatus.setClusterId(vaultSealStatusResult.clusterId);
+        vaultSealStatus.setClusterName(vaultSealStatusResult.clusterName);
+        vaultSealStatus.setInitialized(vaultSealStatusResult.initialized);
+        vaultSealStatus.setMigration(vaultSealStatusResult.migration);
+        vaultSealStatus.setN(vaultSealStatusResult.n);
+        vaultSealStatus.setNonce(vaultSealStatusResult.nonce);
+        vaultSealStatus.setProgress(vaultSealStatusResult.progress);
+        vaultSealStatus.setRecoverySeal(vaultSealStatusResult.recoverySeal);
+        vaultSealStatus.setSealed(vaultSealStatusResult.sealed);
+        vaultSealStatus.setT(vaultSealStatusResult.t);
+        vaultSealStatus.setType(vaultSealStatusResult.type);
+        vaultSealStatus.setVersion(vaultSealStatusResult.version);
+
+        return vaultSealStatus;
+    }
+
+    private VaultHealthStatus healthStatus(boolean isStandByOk, boolean isPerfStandByOk) {
```

Comment:
why did you distinguished `VaultHealthStatus` from `VaultHealth`? we could have added the status code into the `VaultHealthStatus`. did you think that for performance reasons , in the context of healthchecks, it was important to do a `head` instead of a full `get`?

Reply:
Well two reasons, the first is that as you said head is more performant and in case of readiness checks it is important to be as performant as possible and the second is that Vault offers both options in the API. For this reason, I implemented both, so if someone wants to use programmatically is free to use it but in case of health checks use the more performant one.